src/libostree/ostree-repo-static-delta-core.c

```diff
@@ -725,23 +725,52 @@ _ostree_repo_static_delta_dump (OstreeRepo                    *self,
                                 GCancellable                  *cancellable,
                                 GError                       **error)
 {
-  g_autofree char *from = NULL;
-  g_autofree char *to = NULL;
-  if (!_ostree_parse_delta_name (delta_id, &from, &to, error))
-    return FALSE;
+  glnx_autofd int superblock_fd = -1;
 
-  g_autofree char *superblock_path = _ostree_get_relative_static_delta_superblock_path (from, to);
+  if (strchr (delta_id, '/'))
+    {
+      if (!glnx_openat_rdonly (AT_FDCWD, delta_id, TRUE, &superblock_fd, error))
+        return FALSE;
+    }
+  else
+    {
+      g_autofree char *from = NULL;
+      g_autofree char *to = NULL;
+      if (!_ostree_parse_delta_name (delta_id, &from, &to, error))
+        return FALSE;
+
+      g_autofree char *superblock_path = _ostree_get_relative_static_delta_superblock_path (from, to);
+      if (!glnx_openat_rdonly (self->repo_dir_fd, superblock_path, TRUE, &superblock_fd, error))
+        return FALSE;
+    }
 
-  glnx_autofd int superblock_fd = -1;
-  if (!glnx_openat_rdonly (self->repo_dir_fd, superblock_path, TRUE, &superblock_fd, error))
-    return FALSE;
   g_autoptr(GVariant) delta_superblock = NULL;
   if (!ot_variant_read_fd (superblock_fd, 0,
                            (GVariantType*)OSTREE_STATIC_DELTA_SUPERBLOCK_FORMAT,
                            TRUE, &delta_superblock, error))
     return FALSE;
 
   g_print ("Delta: %s\n", delta_id);
+  g_autoptr(GVariant) from_commit_v = NULL;
+  g_variant_get_child (delta_superblock, 2, "@ay", &from_commit_v);
+  g_autofree char *from_commit = NULL;
```

Comment:
Minor: could move this inside the if scope.

Reply:
Hmm no, it's referenced below right?
`      if (!show_one_part (self, swap_endian, from_commit, to_commit, meta_entries, i,`